packages/dart/npt_flutter/lib/features/tray_manager/widgets/tray_manager.dart

```diff
@@ -55,6 +74,17 @@ class _TrayManagerState extends State<TrayManager> with TrayListener, WindowList
           BlocListener<ProfilesRunningCubit, ProfilesRunningState>(
             listener: reloadTray,
           ),
+          BlocListener<SettingsBloc, SettingsState>(
+              listener: reloadTray,
+              // Only call listener when the language changes in settings
+              listenWhen: (prev, next) {
+                if (prev is SettingsLoadedState && next is SettingsLoadedState) {
+                  return prev.settings.language != next.settings.language;
+                }
+                // This may cause some extra reloading (very occasionally, settings shouldn't change often)
+                // but it should catch all of the edge cases
+                return prev is SettingsLoadedState || next is SettingsLoadedState;
+              }),
```

Comment:
Added settings bloc listener for the locale, but it only should  trigger a reload when the language changes (`listenWhen` function)